v3-sdk/minting-position/src/example/Example.tsx

```diff
@@ -0,0 +1,266 @@
+  }
+  // Give approval to the contract to transfer tokens
+  const tokenInApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.in.address,
+    address,
+    NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS
+  )
+  const tokenOutApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.out.address,
+    address,
+    NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS
+  )
+
+  if (
+    tokenInApproval !== TransactionState.Sent ||
+    tokenOutApproval !== TransactionState.Sent
+  ) {
+    return TransactionState.Failed
+  }
+
+  // get pool data
+  const poolConstants = await getPoolConstants()
+  const poolState = await getCurrentPoolState()
+
+  // create Pool abstraction
+  const USDC_DAI_POOL = new Pool(
+    CurrentConfig.tokens.in,
+    CurrentConfig.tokens.out,
+    poolConstants.fee,
+    poolState.sqrtPriceX96.toString(),
+    poolState.liquidity.toString(),
+    poolState.tick
+  )
+
+  // create Position abstraction
+  const position = new Position({
+    pool: USDC_DAI_POOL,
+    liquidity: CurrentConfig.tokens.liquidity,
+    tickLower:
+      nearestUsableTick(poolState.tick, poolConstants.tickSpacing) -
+      poolConstants.tickSpacing * 2,
+    tickUpper:
+      nearestUsableTick(poolState.tick, poolConstants.tickSpacing) +
+      poolConstants.tickSpacing * 2,
+  })
+
+  // get calldata for minting a position
+  const { calldata, value } = NonfungiblePositionManager.addCallParameters(
+    position,
+    {
+      recipient: address,
+      deadline: Math.floor(Date.now() / 1000) + 60 * 20,
+      slippageTolerance: new Percent(50, 10_000),
+    }
+  )
+
+  // build transaction
+  const transaction = {
+    data: calldata,
+    to: NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+    value: value,
+    from: address,
+    maxFeePerGas: MAX_FEE_PER_GAS,
+    maxPriorityFeePerGas: MAX_PRIORITY_FEE_PER_GAS,
+  }
+
+  await sendTransaction(transaction)
+  return TransactionState.Sent
+}
+
+const useOnBlockUpdated = (callback: (blockNumber: number) => void) => {
+  useEffect(() => {
+    const subscription = getProvider()?.on('block', callback)
+    return () => {
+      subscription?.removeAllListeners()
+    }
+  })
+}
+
+const Example = () => {
+  const [tokenInBalance, setTokenInBalance] = useState<string>()
+  const [tokenOutBalance, setTokenOutBalance] = useState<string>()
+  const [positionIds, setPositionIds] = useState<number[]>()
+  const [txState, setTxState] = useState<TransactionState>(TransactionState.New)
+  const [blockNumber, setBlockNumber] = useState<number>(0)
+
+  // Listen for new blocks and update the wallet
+  useOnBlockUpdated(async (blockNumber: number) => {
+    refreshBalances()
+    setBlockNumber(blockNumber)
+  })
+
+  // Update wallet state given a block number
+  const refreshBalances = useCallback(async () => {
+    const provider = getProvider()
+    const address = getWalletAddress()
+    if (address && provider) {
```

Comment:
style: check for nulls and return early instead? Saves indent space and fail fast conventions